### ISSUE TITLE
Recursion option?

### ISSUE THREAD

**Issue by unquietwiki** [kalbhor/MusicRepair#10]
If you're looking for something cool to add to this, an option for directory recursion would be neat. My own attempts at such in programs I've worked on, have been somewhat halting. But I have found some resources that might help you out.

* http://pythoncentral.io/how-to-traverse-a-directory-tree-in-python-guide-to-os-walk/
* https://docs.python.org/3/library/os.html

**Comment by kalbhor** [comment]
I will definitely look into adding an argument that would allow it to recursively fix mp3 files. Thanks for the links!

**Comment by unquietwiki** [comment]
@lakshaykalbhor No problem. :) The thing that's tripped me up, that I hope you won't struggle much with, is the way the data is returned from the tuples. Seems a lot more straightforward in other languages I've used.

**Comment by kalbhor** [comment]
I will take time to add this feature, since I'm busy this week. But I'm open to pull requests.

**Comment by captainwasabi** [comment]
In the mean time, if you are on windows this command seems to be working:

for /r /d %1 in (*) do musicrepair -d "%1"

**Comment by lakshaykalbhor** [comment]
@captainwasabi Could you please post your comment on making it work for windows in the windows issue?

**Comment by brandonb927** [comment]
Ohh I just found this repo via reddit and realized I have been looking for something like this all along. I tried it out and realized that I need recursion (folders in folders in folders of music 😬) so if I have some time I might take a look at implementing this!

**Comment by kalbhor** [comment]
I'd be happy to entertain pull requests :)

**Comment by brandonb927** [comment]
@lakshaykalbhor I have put up a PR if you want to take a look. Let me know if there's anything you want to change with this. I also don't know how the api keys stuff is tested but I'm sure that hasn't changed with this.

**Comment by kalbhor** [comment]
@brandonb927  Thanks for the PR, I have merged it. The only problem that I saw was .decode() doesn't work for python 3. I have added an if statement to differentiate between python 2 and python 3 using a bool in the beginning. I'm not sure whether it's the best way, so I'm open to any change. Also, I'm thankful for you cleaning up the code and logging errors properly. I've learn a lot through your code!
